### PR TITLE
Changes in loading params, fixes, and search objective outcome chagnes

### LINKED ISSUE #50: Change search.py transitions
The search objective currently has 2 outcomes: ['success', 'aborted'] change the transitions to be a little more descriptive ['obj_found', 'obj_not_found'] or something of the nature. Because it is a little unintuitive to realize that we _want_ the search objective to be aborted. Being aborted is successful, it means we're moving on to the next state in the task statemachine

### LINKED ISSUE #90: Move init params of task objects to be loaded from the param server in initObjectives() and not in smach_top
I think we should put task configurations from mission_config in the task's initObjectives() function. Right now we have them a bit scattered about the code but for the most part inside of smach_top.py. This would look something like

```
task_sm = None
        rospy.loginfo("Loading " + task + " state machine")

        visit = False
        # Check for just visiting a task
        if "visit" in task:
            task = task[6:] # trim "visit_"
            visit = True
        prior = genPoseMsg(rospy.get_param("tasks/" + task + "/prior"))
        search_alg = rospy.get_param("tasks/" + task + "/search_alg")

        if visit == True:
            task_sm = VisitTask(prior, search_alg)
        elif task == "start_gate":
            dist_behind_gate = rospy.get_param("tasks/start_gate/dist_behind_gate")
            task_sm = StartGate(prior, search_alg, dist_behind_gate)
```
Would become

```
        task_sm = None
        rospy.loginfo("Loading " + task + " state machine")

        visit = False
        # Check for just visiting a task
        if "visit" in task:
            task = task[6:] # trim "visit_"
            visit = True

        if visit == True:
            task_sm = VisitTask(task_name) # so we know which prior param to grab
        elif task == "start_gate":
            task_sm = StartGate()
```

This gives us more flexibiltiy when adding new configurations, we can just edit the task code instead of editing the task code as well as smach_top in order to grab that new param and pass it in

### PR DESCRIPTION
**Pull request by flamma7**
Closes #90. Closes #50 
- Moved all prior and search alg rosparams to be pulled inside the task themselves for a cleaner smach_top.py and more encapsulated task configuration.
- Changed outcomes of search objective to be more intuitive: "found" and 
"not_found"
- Fixed waypoint server access namespacing in smach_top.py and for all 
objectives
- Moved startgate's "dist_behind_gate" param to the config file
- Consolidated method of parsing priors to the task.py Task class

### CODE REVIEW

**Comment by flamma7** [comment]
Used atom for this commit. It seemed to not like a lot of the spaces in the files.

**Comment by skhadem** [comment]
Lol. I have seen plugins that fix whitespace commits

**Comment by skhadem** [comment]
I guess adding ?w=1 to the end of the url on the diff ignores whitespace so it's easier to look at

File: cusub_cortex/state_machine/src/tasks/start_gate.py [hunk 1]
```diff
@@ -27,40 +27,39 @@ def __init__(self):
...
-        search_alg = rospy.get_param("tasks/start_gate/search_alg")
-        self.search = Search(search_alg, prior, "cusub_cortex/mapper_out/start_gate")
+        search_alg = rospy.get_param("tasks/" + self.name + "/search_alg")
+        self.search = Search(self.getPrior(), search_alg, "cusub_cortex/mapper_out/start_gate")
         self.attack = Attack()
 
     def linkObjectives(self):
         with self: # we are a StateMachine
-            smach.StateMachine.add('Search', self.search, transitions={'aborted':'Attack', 'success':'Search'})
+            smach.StateMachine.add('Search', self.search, transitions={'found':'Attack', 'not_found':'Search'})
```
Comment (skhadem) [inline cusub_cortex/state_machine/src/tasks/start_gate.py:36]: So much better

File: cusub_cortex/state_machine/scripts/smach_top.py [hunk 2]
```diff
@@ -81,9 +72,9 @@ def main():
     Loop through the statemachines in order and link them
     """
     rospy.init_node('smach_top')
-
+    
     # All Objectives depend on the waypoint server so let's wait for it to initalize here
-    wayClient = actionlib.SimpleActionClient('cusub_common/waypoint', waypointAction)
+    wayClient = actionlib.SimpleActionClient('/'+rospy.get_param('~robotname')+'/cusub_common/waypoint', waypointAction)
```
Comment (DrYerzinia) [inline cusub_cortex/state_machine/scripts/smach_top.py:77]: Why did we change this to a global?

File: cusub_cortex/cusub_cortex_bringup/launch/cortex.launch [hunk 3]
```diff
@@ -2,6 +2,8 @@
   <arg name="sub" default="leviathan" />
   <group ns="$(arg sub)/cusub_cortex">
```
Comment (DrYerzinia) [inline cusub_cortex/cusub_cortex_bringup/launch/cortex.launch:3]: I see we are launching a level deeper than I thought hense the global

**Pull request merged**